packages/service-library/src/servicelib/aiohttp/long_running_tasks/client.py

```diff
@@ -0,0 +1,155 @@
+) -> TaskGet:
+    async with session.post(request, json=json) as response:
+        response.raise_for_status()
+        data, error = unwrap_envelope(await response.json())
+    assert not error  # nosec
+    assert data is not None  # nosec
+    task = TaskGet.parse_obj(data)
+    return task
+
+
+async def _wait_for_completion(
+    session: ClientSession,
+    task_id: TaskId,
+    status_url: str,
+    wait_timeout_s: int,
+    wait_interval_s: float,
+) -> AsyncGenerator[TaskProgress, None]:
+    try:
+        async for attempt in AsyncRetrying(
+            wait=wait_fixed(wait_interval_s),
+            stop=stop_after_delay(wait_timeout_s),
+            reraise=True,
+        ):
+            with attempt:
+                async with session.get(status_url) as response:
+                    response.raise_for_status()
+                    data, error = unwrap_envelope(await response.json())
+                    assert not error  # nosec
+                    assert data is not None  # nosec
+                task_status = TaskStatus.parse_obj(data)
+                yield task_status.task_progress
+                if not task_status.done:
+                    raise TryAgain(
+                        f"{task_id=}, {task_status.started=} has "
+                        f"status: '{task_status.task_progress.message}'"
+                        f" {task_status.task_progress.percent}%"
+                    )
+    except TryAgain as exc:
+        # this is a timeout
+        raise asyncio.TimeoutError(
+            f"Long running task {task_id}, calling to {status_url} timed-out after {wait_timeout_s} seconds"
+        ) from exc
+
+
+@retry(**_DEFAULT_AIOHTTP_RETRY_POLICY)
+async def _task_result(session: ClientSession, result_href: str) -> Any:
+    async with session.get(result_href) as response:
+        response.raise_for_status()
+        if response.status != web.HTTPNoContent.status_code:
+            data, error = unwrap_envelope(await response.json())
+            assert not error  # nosec
+            assert data  # nosec
+            return data
+
+
+@retry(**_DEFAULT_AIOHTTP_RETRY_POLICY)
+async def _abort_task(session: ClientSession, abort_href: str) -> None:
+    async with session.delete(abort_href) as response:
+        response.raise_for_status()
+        data, error = unwrap_envelope(await response.json())
+        assert not error  # nosec
+        assert not data  # nosec
+
+
+@dataclass(frozen=True)
+class LRTask:
+    progress: TaskProgress
+    _result: Optional[Coroutine[Any, Any, Any]] = None
+
+    def done(self) -> bool:
+        return self._result is not None
+
+    async def result(self) -> Any:
+        if not self._result:
+            raise ValueError("No result ready!")
+        return await self._result
+
+
+async def long_running_task_request(
+    session: ClientSession,
+    url: URL,
+    json: Optional[RequestBody] = None,
+    wait_timeout_s: int = 1 * _HOUR,
```

Comment:
These timing options are related with the polling interval. I would rather reuse tenacity kwargs. This way the caller has far more flexibility in the way it pools (e.g. it might poll the state with an exponential/random delay etc)

I have some suggestions regarding the name as well below (check the test)

Reply:
ok, I changed these. please have a look.
I kept the client_timeout which makes sense I think